pycaret/internal/plots/helper.py

```diff
@@ -0,0 +1,111 @@
+"""
+This module contains methods that can be used in various plot modules and don't really belong to a specific module
+"""
+
+import numpy as np
+
+
+def leverage_statistic(X: np.ndarray) -> np.array:
+    """
+    Calculates the leverage statistic $h_i$ for all $n$ observations $x_i$ within $X$.
+    $h_{i}=\frac{1}{n}+\frac{\left(x_{i}-\bar{x}\right)^{2}}{\sum_{i^{\prime}=1}^{n}\left(x_{i^{\prime}}-\bar{x}\right)^{2}}$
+    Observations with high leverage have an unusual value for $x_i$.
+
+    Parameters
+    ----------
+    X: np.array
+        A collection of observations $x_i$, where $dim(X)=2$. The first dimension describes the number of observations $n$
+        and the second the number of features.
+
+    Returns
+    -------
+        np.array: A array containing the leverage of each observation $x_i$, hence it has length $n$.
+
+    """
+
+    if X.ndim > 1:
+        x_mean = np.mean(X, axis=0)
+        divider = np.sum([np.power(np.subtract(element, x_mean), 2) for element in X], axis=0)
+        leverage = np.array(
+            [1 / X.shape[0] + np.divide(np.power(np.subtract(element, x_mean), 2), divider) for element in X])
+        return np.sum(leverage, axis=1)
+    else:
+        x_mean = np.mean(X)
+        divider = np.sum([np.power(np.subtract(element, x_mean), 2) for element in X])
+        leverage = np.array(
+            [1 / X.shape[0] + np.divide(np.power(np.subtract(element, x_mean), 2), divider) for element in X])
+        return leverage
+
+
+def calculate_standardized_residual(
+        predicted: np.ndarray,
+        expected: np.ndarray = None,
+        featuresize: int = None
+) -> np.ndarray:
+    """
+    Calculates the standardized residuals $\tilde r_i$ of the predictions to the expectations.
+    $\tilde{r}_{i}=\frac{r_{i}}{\widehat{\sigma} \sqrt{1-\left(\frac{1}{n}+\frac{\left(y_{i}-\bar{y}\right)^{2}}{\sum_{i}^{n}\left(y_{i}-\bar{y}\right)^{2}}\right)}}$
+    where $r_i= y_i - \hat y_i$ and $\hat \sigma$ denoting the estimated standard deviation of the error terms $\epsilon_i\approx r_i$.
```

Comment:
I guess it should be "denote" instead of "denoting" but I am not a nativ english speaker